packages/typespec-client-generator-core/src/types.ts

```diff
@@ -1475,37 +1419,44 @@ function updateUsageOfModel(
   }
   if (type.kind === "model" && options.seenModelNames.has(type)) return; // avoid circular references
   if (type.kind === "array" || type.kind === "dict") {
-    return updateUsageOfModel(context, usage, type.valueType, options);
+    return updateUsageOrAccessOfModel(context, value, type.valueType, options);
   }
   if (type.kind === "union") {
     for (const unionType of type.values) {
-      updateUsageOfModel(context, usage, unionType, options);
+      updateUsageOrAccessOfModel(context, value, unionType, options);
     }
     return;
   }
   if (type.kind === "enumvalue") {
-    updateUsageOfModel(context, usage, type.enumType, options);
+    updateUsageOrAccessOfModel(context, value, type.enumType, options);
     return;
   }
   if (type.kind === "nullable") {
-    updateUsageOfModel(context, usage, type.type, options);
+    updateUsageOrAccessOfModel(context, value, type.type, options);
     return;
   }
   if (type.kind !== "model" && type.kind !== "enum") return;
   options.seenModelNames.add(type);
 
-  const usageOverride = getUsageOverride(context, type.__raw as any);
-  if (usageOverride) {
-    type.usage |= usageOverride | usage;
+  if (typeof value === "number") {
+    // usage set
+    type.usage |= value;
   } else {
-    type.usage |= usage;
+    // access set
+    if (!type.__accessSet || type.access !== "public") {
+      type.access = value;
+      type.__accessSet = true;
+    }
+    if (options.isOverride) {
```

Comment:
I think another option here so we don't need `__accessSet` is still setting it to some kind of default, like undefined, and at the end before we return, all `undefined`s can be set to `"public"`, since we're already iterating over them

Reply:
i tried to do that at first. but it should either make the interface access to be | undefined or use as magic to force set to the type. maybe the later one is better?

Reply:
ok no worries, that's a good point about how we don't want `undefined` to be a valid value for `access`